packages/subproviders/src/subproviders/mnemonic_wallet.ts

```diff
@@ -0,0 +1,138 @@
+            ...this._derivedKey,
+            derivationBasePath: this._derivationBasePath,
+        };
+    }
+    /**
+     * Retrieve the accounts associated with the mnemonic.
+     * This method is implicitly called when issuing a `eth_accounts` JSON RPC request
+     * via your providerEngine instance.
+     * @param numberOfAccounts Number of accounts to retrieve (default: 10)
+     * @return An array of accounts
+     */
+    public async getAccountsAsync(
+        numberOfAccounts: number = walletUtils.DEFAULT_NUM_ADDRESSES_TO_FETCH,
+    ): Promise<string[]> {
+        const derivedKeys = walletUtils.calculateDerivedHDKeys(this._derivedKey, numberOfAccounts);
+        const accounts = _.map(derivedKeys, 'address');
+        return accounts;
+    }
+
+    /**
+     * Signs a transaction with the from account specificed in txParams.
+     * If you've added this Subprovider to your  app's provider, you can simply send
+     * an `eth_sendTransaction` JSON RPC request, and * this method will be called auto-magically.
+     * If you are not using this via a ProviderEngine instance, you can call it directly.
+     * @param txParams Parameters of the transaction to sign
+     * @return Signed transaction hex string
+     */
+    public async signTransactionAsync(txParams: PartialTxParams): Promise<string> {
+        const privateKeyWallet = this._privateKeyWalletForAddress(txParams.from);
+        const signedTx = privateKeyWallet.signTransactionAsync(txParams);
+        return signedTx;
+    }
+    /**
+     * Sign a personal Ethereum signed message. The signing account will be the account
+     * associated with the provided address.
+     * If you've added the MnemonicWalletSubprovider to your app's provider, you can simply send an `eth_sign`
+     * or `personal_sign` JSON RPC request, and this method will be called auto-magically.
+     * If you are not using this via a ProviderEngine instance, you can call it directly.
+     * @param data Message to sign
+     * @param address Address to sign with
+     * @return Signature hex string (order: rsv)
+     */
+    public async signPersonalMessageAsync(data: string, address: string): Promise<string> {
+        const privateKeyWallet = this._privateKeyWalletForAddress(address);
+        const sig = await privateKeyWallet.signPersonalMessageAsync(data, address);
+        return sig;
+    }
+    private _privateKeyWalletForAddress(address: string): PrivateKeyWalletSubprovider {
+        const derivedKey = this._findDerivedKeyByPublicAddress(address);
+        const privateKeyHex = derivedKey.hdKey.privateKey.toString('hex');
+        const privateKeyWallet = new PrivateKeyWalletSubprovider(privateKeyHex);
+        return privateKeyWallet;
+    }
+    private _findDerivedKeyByPublicAddress(address: string): DerivedHDKey {
+        if (_.isUndefined(address) || !addressUtils.isAddress(address)) {
+            throw new Error(WalletSubproviderErrors.FromAddressMissingOrInvalid);
```

Comment:
This error is thrown at the wrong level. Needs to live at the function that receives the `from` address.